vizro-core/docs/pages/user-guides/data.md

```diff
@@ -48,17 +48,17 @@ A static data source is the simplest way to send data to your dashboard and shou
 
 You can directly supply a pandas DataFrame into components such as [graphs](graph.md), [tables](table.md), and [figures](figure.md).
 
-The below example uses the Iris data saved to a file `iris.csv` in the same directory as `app.py`. This data can be generated using `px.data.iris()` or [downloaded](../../assets/user_guides/data/iris.csv).
+The examples on this page use the iris dataset, which can be generated by calling `px.data.iris()`. You can alternatively download the iris data from [Vizro's GitHub repository](https://raw.githubusercontent.com/mckinsey/vizro/main/vizro-core/docs/assets/user_guides/data/iris.csv), then store it and read it from file.
```

Comment:
Found it: https://github.com/mckinsey/vizro/pull/398#discussion_r1559668555

Reply:
Thank you! So @antonymilne said that he...

> decided that wasn't clear enough because it's too far removed from a "real" working example where the data can change. So better to use pd.read_csv even though that means downloading a file to get the example working.

The problem (for me) is that when the examples are in py.cafe, there's no easy way to bundle the file with the snippet for the plugin to build the py.cafe example.

Solutions included:

1. What I've done -- get the dataframe (understand why this isn't the best for the example)
2. Download the CSV in code each time

As a non coder, I'd steal this from Petar for my LinkedIn example (but obviously substitute the Iris data URL). Is this the best way to do it?

```py
# Define the URL of the raw CSV file
url = "https://raw.githubusercontent.com/stichbury/vizro_projects/main/linkedin_project/Connections.csv"

# Fetch the CSV file
response = requests.get(url)
response.raise_for_status()  # Ensure we notice bad responses

# Load the CSV data into a pandas DataFrame
from io import StringIO
csv_data = StringIO(response.text)
df = pd.read_csv(csv_data)

```

3. Do what I've had to do for a few other examples (mostly where we use CSS) which is pre-build py.cafe projects and link to those (rather than build them dynamically with the mkdocs plugin). The disadvantage of this is that we then have the code in 2 places -- in the docs and in my project. Any time we change the code we have to change the project (and if I'm not available, it means someone else has to remake the project and change the link to it in the docs).

There are also a lot of examples in this file so if we do that, it's going to take a while to set up all the projects 😬 

My preference was for 1. but I'm happy to consider 2. and explain it in the docs. Really not happy about 3.

WDYT @maxschulz-COL and @huong-li-nguyen  ?

Reply:
I was never so against the `px.data.iris()` route - I see the "problem" but I'd much rather have a smooth flexible example than someone downloading the iris data and saving it to disc.

So my vote goes to 1 💪

Reply:
P.S. We could add a line below in the first or in all examples which says

```python
# df = pd.read_csv(iris.csv) <-- could also be a call to load from disc
```

Reply:
Hmm..tricky.. ultimately it's a choice between "ensuring it works seamlessly in py.Cafe" and "providing a more realistic example." Given that the data manager is generally a more advanced how-to guide, I would have thought the latter is more important. Additionally, it doesn't make much sense to use `px.data.iris()` if the primary goal is to demonstrate dynamic data updates 😬. 

Note that we prefer using the `px.data.iris()` call for all other examples, but in this case, it doesn't seem to fit. 

Perhaps good to ask @antonymilne on this. I'm okay with either option 👍.

Reply:
@stichbury We tried to align between the three of us - the current verdict is, let's not convert this page to py.cafe until either or all of the following have been ressolved
- we use a proper API calling example instead of loading from disk
- we can add files to py.cafe
- we can have proper code block annotations